### DISCUSSION TITLE
Error in coc.nvim: Cannot find module '@tauri-apps/api/tauri' (TS2792)

### DISCUSSION THREAD

**Discussion by Andrew15-5** [tauri-apps/tauri#7965] [Q&A] [answered]
I just wanted to try and make an empty app with tauri, but the first thing I got in my IDE (Neovim + coc.nvim + coc-tsserver with no specific configuration) is this:

<details><summary>screenshot</summary>

![image](https://github.com/tauri-apps/tauri/assets/37143421/e6e5c2f8-0558-4bc3-8522-9d2db160c148)

</details>

```ts
import { useState } from "react";
import reactLogo from "./assets/react.svg";
import { invoke } from "@tauri-apps/api/tauri";
import "./App.css";
```

```log
Cannot find module '@tauri-apps/api/tauri'. Did you mean to set the 'moduleResolution' option to 'node', or to add aliases to the 'paths' option? (tsserver 2792)
```

This (thankfully) obviously fixes the error:

```ts
import { invoke } from "../node_modules/@tauri-apps/api/tauri";
```

The problem is that in both `tsconfig.json` and `tsconfig.node.json` there is a `"moduleResolution": "bundler"` line. How can I resolve this problem?

**Comment by Andrew15-5** [comment, marked as answer]
I did resolve this by writing in `tsconfig.json` > `compilerOptions`:

```json
    "paths": {
      "@tauri-apps/api/tauri": ["./node_modules/@tauri-apps/api/tauri"]
    },
```

But this doesn't explain why I get the error in the first place (and not everybody else).

**Update 1**:
I run an Ubuntu docker container and with minimal copying of my existing setup this and other errors are not present. Now I need to figure out why is that.

**Update 2**:
I think I know what was wrong. I had to run `CocUpdate` and wait for coc plugins to update (including `tsserver`). After this, all errors magically disappear.

**Reply by CristhianMotoche** [reply]
Updating tsserver solved my issue. Thanks!

**Comment by xmsz** [comment]
any update?

**Reply by Andrew15-5** [reply]
https://github.com/tauri-apps/tauri/discussions/7965#discussioncomment-7199637

**Reply by l1xnan** [reply]
[.changes/api-primitives.md](https://github.com/tauri-apps/tauri/blob/dev/.changes/api-primitives.md)
```
---
'@tauri-apps/api': 'patch:breaking'
---

Changed `tauri` module to `primitives` and removed the undocumented `invoke` export from the root module.
```

**Comment by anver** [comment]
make changes to tsconfig.json add

```json
"paths": {
      "@tauri-apps/api/tauri": ["./node_modules/@tauri-apps/api/tauri"]
    },
```

Then where you need invoke function

```ts
import { invoke } from "@tauri-apps/api/primitives"
```

**Comment by metrue** [comment]
I resolved it by running `CocUpdate` for vim. my current version coc.vim is `0.0.82`
